src/metabase/models/params/chain_filter.clj

```diff
@@ -546,20 +546,27 @@
    constraints :- [:maybe ConstraintsMap]
    & options]
   (assert (even? (count options)))
-  (let [{:as options} options]
-    (if-let [v->human-readable (human-readable-remapping-map field-id)]
-      ;; This is for fields that have human-readable values defined (e.g. you've went in and specified that enum
-      ;; value `1` should be displayed as `BIRD_TYPE_TOUCAN`). `v->human-readable` is a map of actual values in the
-      ;; database (e.g. `1`) to the human-readable version (`BIRD_TYPE_TOUCAN`).
-      (-> (unremapped-chain-filter field-id constraints options)
-          (update :values add-human-readable-values v->human-readable))
-      (if (use-cached-field-values? field-id)
-        (cached-field-values field-id constraints options)
-        (if-let [remapped-field-id (remapped-field-id field-id)]
-          ;; This is Field->Field remapping e.g. `venue.category_id `-> `category.name `;
-          ;; search by `category.name` but return tuples of `[venue.category_id category.name]`.
-          (unremapped-chain-filter remapped-field-id constraints (assoc options :original-field-id field-id))
-          (unremapped-chain-filter field-id constraints options))))))
+  (let [{:as options}         options
+        v->human-readable     (delay (human-readable-remapping-map field-id))
+        the-remapped-field-id (delay (remapped-field-id field-id))]
+    (cond
+     ;; This is for fields that have human-readable values defined (e.g. you've went in and specified that enum
+     ;; value `1` should be displayed as `BIRD_TYPE_TOUCAN`). `v->human-readable` is a map of actual values in the
+     ;; database (e.g. `1`) to the human-readable version (`BIRD_TYPE_TOUCAN`).
+     (some? @v->human-readable)
```

Comment:
Is it really worth making `v->human-readable` a delay if you always end up dereffing it?